### DISCUSSION TITLE
SSL parameters for ui.run since 1.1.10 release?

### DISCUSSION THREAD

**Discussion by pythonormen** [zauberzeug/nicegui#575] [Q&A] [answered]
If I understand it correctly, then it should, as per release 1.1.10, be possible to add SSL parameters inside ui run?

Any plan to update the documentation or if possible show an example with the required parameters to add? 

Thank you so much

**Comment by falkoschindler** [comment, marked as answer]
Thanks for bringing this up, @pythonormen. Our documentation is, indeed, a bit sparse...
At https://nicegui.io/reference#ui_run we mention:
> kwargs: additional keyword arguments are passed to uvicorn.run

This refers to https://www.uvicorn.org/#uvicornrun where parameters like
```
--ssl-keyfile TEXT              SSL key file
--ssl-certfile TEXT             SSL certificate file
```
are documented. When running uvicorn programmatically (as we do), they translate to `ui.run(ssl_keyfile=..., ssl_certfile=...)` (underscore instead of dash) like shown [here](https://stackoverflow.com/a/70542746/3419103) for example.

So in order to keep it generic and to avoid the need for keeping our documentation in sync with uvicorn, we only introduced `kwargs` and let you pass to uvicorn whatever you like. I hope this works for you nevertheless.
